### ISSUE TITLE
[Feature Request] Pure rust implementation

### ISSUE THREAD

**Issue by juarezr** [fernandobatels/rsfbclient#13]
### [Feature Request] Pure rust implementation

1. Actually `rsfbclient` requires `fbclient` library available for low level protocol handling.
2. It's a lot of work, but it's possible to develop this library in pure `Rust `.
3. For example, [Jaybird](https://firebirdsql.github.io/jaybird-manual/jaybird_manual.html#_jaybird_architecture) rewrote the protocol in [pure java](https://github.com/FirebirdSQL/jaybird/tree/master/src/main/org/firebirdsql/gds).
4. The main pros are:
    1. Assure that library is safe from memory leakis and race conditions, granted by Rust borrow checker.
    2. No need for unsafe code.
    3. No need for deploying or static linking `fbclient` library.

Of course it's better get it working and stable using `fbclient` library first and after explore this possibility.

**Comment by fernandobatels** [comment]
It's a possibility, but demands a lots of work.

When I started this project, I tried to make with pure Rust code, but the firebird protocol isn't widely documented, especially the differences between firebird versions. For this reason, I focused in use the fbclient.

And the firebird community always use the fbclient instead of write from scratch. The jaybird is an exception.

-------

I think if we cover the entire lib with good tests, we can think in change the core to use only rust code.

------

If you found more documentation about the firebird protocol, feel free to share here.
